core/src/main/java/org/frankframework/jdbc/dbms/TransactionalDbmsSupportAwareDataSourceProxy.java

```diff
@@ -0,0 +1,173 @@
+			info.append("driver [").append(metadata.get("driver")).append(CLOSE);
+			info.append("driver version [").append(metadata.get("driver-version")).append(CLOSE);
+		}
+
+		if (getTargetDataSource() instanceof OpenManagedDataSource) {
+			OpenManagedDataSource targetDataSource = (OpenManagedDataSource) getTargetDataSource();
+			GenericObjectPool pool = targetDataSource.getPool();
+			getGenericObjectPoolInfo(pool, info);
+		} else if (getTargetDataSource() instanceof org.apache.commons.dbcp2.PoolingDataSource) {
+			OpenPoolingDataSource dataSource = (OpenPoolingDataSource) getTargetDataSource();
+			GenericObjectPool pool = dataSource.getPool();
+			getGenericObjectPoolInfo(pool, info);
+		} else if (getTargetDataSource() instanceof BasicDataSource) { // Tomcat instance
+			BasicDataSource dataSource = (BasicDataSource) getTargetDataSource();
+			info.append("Tomcat Pool Info: ");
+			if (dataSource != null) {
+				info.append("maxIdle [").append(dataSource.getMaxIdle()).append(CLOSE);
+				info.append("minIdle [").append(dataSource.getMinIdle()).append(CLOSE);
+				info.append("maxTotal [").append(dataSource.getMaxTotal()).append(CLOSE);
+				info.append("numActive [").append(dataSource.getNumActive()).append(CLOSE);
+				info.append("numIdle [").append(dataSource.getNumIdle()).append(CLOSE);
+				info.append("testOnBorrow [").append(dataSource.getTestOnBorrow()).append(CLOSE);
+				info.append("testOnCreate [").append(dataSource.getTestOnCreate()).append(CLOSE);
+				info.append("testOnReturn [").append(dataSource.getTestOnReturn()).append(CLOSE);
+			}
+		} else if (getTargetDataSource() instanceof PoolingDataSource) { // BTM instance
+			PoolingDataSource dataSource = (PoolingDataSource) getTargetDataSource();
+			info.append("BTM Pool Info: ");
+			if (dataSource != null) {
+				info.append("maxPoolSize [").append(dataSource.getMaxPoolSize()).append(CLOSE);
+				info.append("minPoolSize [").append(dataSource.getMinPoolSize()).append(CLOSE);
+				info.append("totalPoolSize [").append(dataSource.getTotalPoolSize()).append(CLOSE);
+				info.append("inPoolSize [").append(dataSource.getInPoolSize()).append(CLOSE);
+			}
+		}
+
+		info.append(" datasource [").append(obtainTargetDataSource().getClass().getName()).append("]");
+		return info.toString();
+	}
+
+	private static void getGenericObjectPoolInfo(GenericObjectPool pool, StringBuilder info) {
+		if (pool == null) {
+			return;
+		}
+		info.append("DBCP2 Pool Info: ");
+		info.append("maxIdle [").append(pool.getMaxIdle()).append(CLOSE);
+		info.append("minIdle [").append(pool.getMinIdle()).append(CLOSE);
+		info.append("maxTotal [").append(pool.getMaxTotal()).append(CLOSE);
+		info.append("numActive [").append(pool.getNumActive()).append(CLOSE);
+		info.append("numIdle [").append(pool.getNumIdle()).append(CLOSE);
+		info.append("testOnBorrow [").append(pool.getTestOnBorrow()).append(CLOSE);
+		info.append("testOnCreate [").append(pool.getTestOnCreate()).append(CLOSE);
```

Comment:
I really feel like all this info does not belong in this class.

Reply:
I was thinking that since we extend some classes to give access to underlying pools or datasources, perhaps we can make the `OpenManagedDataSource` and the `OpenPoolingDataSource` describe these pools and datasources instead? And then all this can be removed from here.

Reply:
These lines apply to the `GenericObjectPool` from the `commons-pool2` library. Moving it to both `Open*DataSource` classes is not nice. Adding a new class, for this purpose also is a bit too much, but perhaps the best option then.